src/v/cluster/types.h

```diff
@@ -1262,4 +1262,19 @@ struct adl<cluster::topic_properties> {
     void to(iobuf&, cluster::topic_properties&&);
     cluster::topic_properties from(iobuf_parser&);
 };
+
+template<typename T>
+struct adl<cluster::property_update<T>> {
+    void to(iobuf& out, cluster::property_update<T>&& update) {
+        reflection::serialize(out, std::move(update.value), update.op);
+    }
+
+    cluster::property_update<T> from(iobuf_parser& parser) {
+        return {
+          .value = reflection::adl<T>{}.from(parser),
+          .op = reflection::adl<cluster::incremental_update_operation>{}.from(
+            parser),
```

Comment:
1. let's not do this. in the best case scenario it ties correctness to the order of the fields in the structure (a huge reason for moving away from reflection). But even if we believe that the order of initialization is correct, do we know  that the order of evaluation of the LHS of the assignment is the same? It isn't safe in cases like foo(read(), read()).

Reply:
followed up too to the other discussion with @BenPope related to this topic here https://github.com/redpanda-data/redpanda/pull/5033#discussion_r892813937